lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -612,10 +632,23 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
                              Desc.getASTFile());
   };
 
+  static Optional<ASTSourceDescriptor> getClangModule(const ModuleDecl &M) {
+    for (auto *FU : M.getFiles()) {
+      if (auto *SAF = dyn_cast_or_null<SerializedASTFile>(FU)) {
+        if (auto *ShadowedModule = SAF->getShadowedModule())
+          if (auto Desc = getClangModule(*ShadowedModule))
+            return Desc;
```

Comment:
This part seems unnecessary. What are you using it for?

Reply:
There is a code path that calls getOrCreateModule(ModuleDecl::ImportedModule IM) with a ModuleDecl that is a SerializedASTFile that is shadowing the ClangModuleUnit. This step makes sure the Clang module debug info is created instead of description of the Swift shadow module. I've seen this happen for the Objective-C and Foundation modules. Were those the SDK overlays and is the new behavior actually more accurate? In any case, this was necessary to make this an NFC change.

Reply:
Hm. I *do* think the new behavior is more correct; the current behavior implies that Swift declarations in the overlays are coming from the Clang module, which they aren't. (Yes, those are overlays.) However, in terms of minimizing risk for 5.0, it's certainly better to make this an NFC change.

The other thing that bothers me is the promotion of `getShadowedModule` to general use. That's supposed to be an implementation detail of sorts, which `findUnderlyingClangModule` being the more semantic entry point. On the other hand, I bet we're already looking through it in several other places. One way to make me feel better about this (but not really change much in practice) would be to make this a static API on SerializedModuleLoader instead (and not add it to SerializedASTFile), so that it's at least clear whose abstraction we're breaking.

We are on a time crunch, though, so if you think it's important to get this in and do that as a follow-up, then okay.

Reply:
That's the nice thing; I don't need to expose the getShadowedModule() call on the swift/master branch at all. I'll update the testcase instead.

I'll leave it as is on https://github.com/apple/swift/pull/22316 though to minimize the change.